FIX #27536 VAT of the buyer's country while SERVICE_ARE_ECOMMERCE_200238EC option is activated

comaiteseb opened the pull request:
VAT of the buyer's country not applied if it is an individual while the SERVICE_ARE_ECOMMERCE_200238EC option is activated
Just a logic issue line 6514 and reduce VAT list in document to the buyer's country VAT

frederic34 committed:
replace old parameter facid by id (#27416)

* replace old parameter facid by id

* Update facture.class.php

eldy committed:
Fix trans

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch '19.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Debug v20

noec764 committed:
Feat develop add hook on print bar code page (#27390)

* NEW: Add hook on printsheet barcode page

* Remove space at the end of line for Travis validation on printsheet.php

---------

Co-authored-by: Noé <noe@scopen.fr>
Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

W1W1-M committed:
Fixed method filter argument type. Cleaned code.

eldy committed:
NEW Add method getPublicImageOfObject() for website module

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Rename into getImagePublicURLOfObject

eldy committed:
Add method getNbOfImagePublicURLOfObject()

frederic34 committed:
remove use not used (#27463)

* remove use not used

* Update website.lib.php

frederic34 committed:
fix phpstan (#27440)

frederic34 committed:
fix Undefined variable $mode (#27451)

mdeweerd committed:
Fix: Disable execute bit in git + correct shebangs (#27450)

# Fix: Execute bits in git and shebangs

Some files had the execute bit improperly set to enabled.
Some scripts did not have a correct shebang.

frederic34 committed:
clean code and fix (#27446)

* clean code and fix

* clean code and fix

mdeweerd committed:
Qual: Spelling outside htdocs (#27448)

* Qual: Spelling outside htdocs

# Qual: Fix spelling of files not in the htdocs subdirectory.

Highlight:
- Change in email test from unvalid to invalid that should give the same test result.

* Update Dolibarr-soapui-project.xml

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

frederic34 committed:
try to run action on push to have always valid cache (#27445)

frederic34 committed:
fix phpstan (#27444)

frederic34 committed:
fix phpstan (#27442)

* fix phpstan

* Update api_products.class.php

frederic34 committed:
fix phpstan (#27441)

mdeweerd committed:
Qual: Spelling htdocs/[q-z]* (#27447)

# Qual: Fix spelling htdocs/[q-z]*

Probably fixes a bug:

- htdocs/webservices/server_thirdparty.php
  No other occurence of adress found.
  -'adress'=>->adress,- +'address'=>->address,+

eldy committed:
FIX #27439 but using the opposite direction. int is bette as we can evolve to add a third value in a future.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

nat-itopie committed:
FIX Error creating a thirdparty (#27438)

The mod_codeclient_* constructors now expect an argument.
See #27373

W1W1-M committed:
Fixed method filter argument type. Cleaned code. Added contributor details. (#27464)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

thibdrev committed:
qual: move into profid.lib.php the code checking the idprof1 syntax for BE, DZ and PT (#27443)

* qual: Update profid.lib.php - move functions checking profid from societe.class.php

- One function checking the Tax Identification Number for PT (called NIF)
- One function checking the Tax Identification Number for DZ (called NIF)
- One function checking the Tax Identification Number for BE (called NN)

* qual: Update societe.class.php - use the isValidTinXX() functions from profid.lib.php

isValidTinForPT($this->idprof1)
isValidTinForDZ($this->idprof1)
isValidTinForBE($this->idprof1)

* typo

mdeweerd committed:
Qual: Fix trailing whitespace in yaml file. (#27466)

# Qual: Fix trailing whitespace in yaml file.

Yaml check tools complain about trailing whitespace, so I fix them.
pre-commit (#27392) has rules to fix them automatically (locally).

frederic34 committed:
remove unused code (#27437)

mdeweerd committed:
Qual: Spelling htdocs/p[b-z]* (#27436)

# Qual: Fix spelling htdocs/p[b-z]*

Highlight:

Probably fixes a bug:
htdocs/product/stock/replenish.php: empty(->variants->eabled) corrected to ...enabled.
'eabled' could not be found elsewhere in the codebase.

lamrani002 committed:
Fix nb of evaluation and display speciale char (#27434)

* Fix nb of evaluation and display speciale char

* Update hrm_evaluation.lib.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

frederic34 committed:
fix phpdoc (#27433)

frederic34 committed:
fix phpdoc (#27432)

frederic34 committed:
fix phpdoc (#27431)

frederic34 committed:
fix phpdoc (#27430)

frederic34 committed:
fix phpdoc (#27428)

frederic34 committed:
add info in service tooltip (#27423)

mdeweerd committed:
Qual: Spelling htdocs/[l-o]* & htdocs/pa* (#27435)

# Qual: Fix spelling htdocs/[l-o]* & htdocs/pa*

Mostly comments, some local variables changed and translations to avoid
automatic fixes.

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Fix phpcs

eldy committed:
Fix phpstan

frederic34 committed:
fix phpdoc (#27429)

frederic34 committed:
clean code (#27421)

Hystepik committed:
Fix funnel of prospection to have a minimum value (#27426)

Co-authored-by: Hystepik <lmarcouiller@nltechno.com>

Daoud-mohamed committed:
NEW: add position column in category table  (#27425)

* add position column in category table

* Update 19.0.0-20.0.0.sql

* Update llx_categorie.sql

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Fix method undef

frederic34 committed:
fix incorrect case and remove exclude check (#27449)

frederic34 committed:
add hook getBuyPrice (#27420)

eldy committed:
Fix phpcs

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix phpcs

eldy committed:
Fix phpstan

mdeweerd committed:
Qual: Spelling for htdocs/[a-b]* (#27422)

Mostly comments, one variable changed from misspelling to .
Also changed CNAT to C.N.A.T in comment to avoid fix from CNAT to CAN'T.

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

thibdrev committed:
qual: move the function checking the syntax of Spanish Tax Identification Numbers (CIF, NIF, NIE) to profid.lib.php (#27472)

* qual: Move the function checking Spanish Tax Identification Numbers from societe.class.php to profid.lib.php

Create the function isValidTinForES($str)

* qual: Update societe.class.php using function isValidTinForES($str) from profid.lib.php

* phpcs

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

mdeweerd committed:
Qual: Ignore most shellcheck annotations, fix some (#27452)

# Qual: Ignore most shellcheck annotations, fix some

Most shellcheck annotations are ignored - this is to make sure that ci
can work for existing scripts.

Some fixes were applied.

pre-commit script was beautyfied by temporarily suppressing the IFS= expression on 2 lines.

eldy committed:
Qual: methods CRUD must have $user as first parameter

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
FIX SQL error

eldy committed:
Merge branch '19.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix look and feel v19

eldy committed:
FIX Filter on rate

eldy committed:
Fix phan report

eldy committed:
Fix phan report

JonBendtsen committed:
Using the same timezones for Argentina which is mentioned in https://www.php.net/manual/en/timezones.america.php (#27473)

Co-authored-by: Jon Bendtsen <xcodeauthor@jonb.dk>

eldy committed:
Merge branch '19.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Style apstats

eldy committed:
Add percent

eldy committed:
Fix percent

eldy committed:
Add bar graph

eldy committed:
css

eldy committed:
Fix bargraph

eldy committed:
Better graph bar

eldy committed:
css

eldy committed:
apstats

eldy committed:
Prepare apstats for lines of code

eldy committed:
Fix apstats

eldy committed:
Enhance script for git stats

eldy committed:
Dev

eldy committed:
Rename shell tools

eldy committed:
Fix params

eldy committed:
Fix message

eldy committed:
Enhance apstats

eldy committed:
Fix apstats

eldy committed:
apstats

eldy committed:
apstats

eldy committed:
apstats

eldy committed:
Fix css

eldy committed:
apstats

eldy committed:
apstats

eldy committed:
apstats

eldy committed:
css

eldy committed:
apstats

frederic34 committed:
fix phpstan (#27489)

hansemschnokeloch committed:
Qual enhance API description (#27487)

frederic34 committed:
fix phpstan (#27488)

frederic34 committed:
fix phpstan (#27483)

* fix phpstan

* fix phpstan

hansemschnokeloch committed:
Enhance API description (#27485)

frederic34 committed:
fix phpstan (#27482)

frederic34 committed:
fix phpstan (#27481)

hansemschnokeloch committed:
Qual enhance API description (#27490)

JonBendtsen committed:
Updated names for timezones (#27478)

* Updated names for 2 US timezones where PHP list the old name in "do not use"
with the new name which PHP lists in the "America" category

https://www.php.net/manual/en/timezones.others.php
https://www.php.net/manual/en/timezones.america.php

* 2 more timezones listed as deperated

* Listed in https://www.php.net/manual/en/timezones.africa.php

* listed as deperated in https://www.php.net/manual/en/timezones.others.php

* new name in https://www.php.net/manual/en/timezones.america.php

* new name from https://www.php.net/manual/en/timezones.america.php

* also deprecated https://www.php.net/manual/en/timezones.others.php

* Godthab changed name to Nuuk

https://data.iana.org/time-zones/tzdb/NEWS

* deprecated https://www.php.net/manual/en/timezones.others.php

* apparently deprecated https://www.php.net/manual/en/timezones.others.php

* also deprecated https://www.php.net/manual/en/timezones.others.php

* old australian timezone names https://www.php.net/manual/en/timezones.others.php

* new australian timezones https://www.php.net/manual/en/timezones.australia.php

* all 4 Brazil timezones are deprecated https://www.php.net/manual/en/timezones.others.php

* canada and CET deprecated https://www.php.net/manual/en/timezones.others.php

* deprecated https://www.php.net/manual/en/timezones.others.php

* added the missing timezones from https://www.php.net/manual/en/timezones.php

---------

Co-authored-by: Jon Bendtsen <xcodeauthor@jonb.dk>

frederic34 committed:
fix phpstan (#27475)

* fix phpstan

* Update mod_codeproduct_leopard.php

frederic34 committed:
fix phpstan (#27476)

mdeweerd committed:
Qual: Fix line endings of yaml files for yamllint. (#27479)

# Qual: Fix line endings of yaml files for yamllint.

yamllint (included in the pre-commit setup) has some messages about the line-endings of yml files.
This fixes that.

mdeweerd committed:
Qual: Fix typing hints for DoliDB (#27484)

# Qual: Fix typing hints for DoliDB

This should help the static tools for type and method checking.

FHenry committed:
feat: add hook in main.inc.php to override the manifest.json (#27459)

* feat: add hook in main.inc.php to override the manifest.json

* feat: add hook in main.inc.php to override the manifest.json

* reveiw

eldy committed:
Fix git tools

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

hansemschnokeloch committed:
QUAL Enhance API description - Documents (#27492)

* qual enhance api description

* Update api_documents.class.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Fix phpunit

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

mdeweerd committed:
Qual: Spelling htdocs/core (#27491)

Highlights:
- initialY in pdf_standard.modules.php was unused
  - remove (otherwise -> initially) (+same in other file)

mdeweerd committed:
Qual: Spelling htdocs/[d-k]* (#27427)

Highlights:

- htdocs/install/doctemplates/websites/website_template-restaurant/containers/page249.tpl.php

  Potential functional change (fix) because toekn was changed to token:
   `<input type=hidden name=toekn`

- htdocs/install/upgrade2.php
  $filles changed to $children

- htdocs/intracommreport/class/intracommreport.class.php
  Local $enveloppe changed to $envelope.

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Close #27496

frederic34 committed:
fix phpstan (#27497)

* fix phpstan

* Update reception.class.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

mdeweerd committed:
Improve 2 of the github scripts (#27493)

mdeweerd committed:
Qual: Spelling fixes that newly appeared. (#27495)

* Qual: Spelling fixes that newly appeared.

# Qual: Fix spelling erros that appeared since the bulk updates.

These issues mostly appeared since the bulk updates (a few ones were postponed
to make sure no real spelling errors were missed.)

* Update modulebuilder.lib.php

* Update pdf_standard.modules.php

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

fbagnol committed:
NEW Add createInvoiceFromContract to API invoice (#27277)

* Add createFromContract to API invoice

* Update invoice : if cond_reglement_id change then update date_lim_reglement

* Space

* indent

---------

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

eldy committed:
Fix phpcs

eldy committed:
Try to merge phpstan.neon and phpstan_action.neon

eldy committed:
Avoid spell warning

mdeweerd committed:
Qual: Add pre-commit & codespell (#27392)

Co-authored-by: Laurent Destailleur <eldy@destailleur.fr>

mdeweerd committed:
Fix: pre-commit: Skip branch check in ci (+used sed in codespell exceptions generation) (#27499)

* Use 'sed' instead of 'perl' (sed is more common)

* Skip branch check in ci

eldy committed:
Clean hooks

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
fix phpstan (#27501)

frederic34 committed:
fix phpstan (#27502)

eldy committed:
Add comment

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Try to move files from root in dedeicated dir

eldy committed:
Fix precommit file

eldy committed:
Merge branch 'develop' into develop

eldy committed:
Fix precommit

eldy committed:
Merge branch 'develop' into develop

eldy committed:
Try to move files from root in dedeicated dir (#27503)

eldy committed:
Merge branch 'Dolibarr:develop' into develop

eldy committed:
Merge pull request #27504 from eldy/develop

test

eldy committed:
Try move setup in config file

eldy committed:
Update pyproject.toml

eldy committed:
Update .pre-commit-config.yaml

eldy committed:
Update .pre-commit-config.yaml

eldy committed:
Update .pre-commit-config.yaml

eldy committed:
Merge pull request #27505 from eldy/develop

Test move option in the precommit config file

mdeweerd committed:
Fix: tools: Small fix with regards to path issue detection

eldy committed:
Merge pull request #27506 from mdeweerd/fix/notfound-message

Fix: tools: Small fix with regards to path issue detection

eldy committed:
Try to merge the 2 phpstan files

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' into develop-fix-27221

eldy committed:
Fix input lost

eldy committed:
Fix phpcs

eldy committed:
Merge branch 'develop' into develop-fix-27221

mdeweerd committed:
Qual: Update logToCs to extract yamllint messages

# Qual: Update logToCs to extract yamllint messages

logToCs did not identify the yamllint messages in the pre-commit action.
This is now added

eldy committed:
Merge pull request #27507 from mdeweerd/logToCs/yamllint-extraction

Qual: Update logToCs to extract yamllint messages

frederic34 committed:
fix phpstan

eldy committed:
Add migration v20 in travis script

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #27222 from glu000/develop-fix-27221

New Feature: E-Mailcollector: Save attachments on operation "recordevent" for tickets

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

mdeweerd committed:
Fix: github_lines_byuser.sh - resolve issue with iconv

# Fix: github_lines_byuser.sh - resolve issue with iconv

iconv could not convert certain characters and truncated the output.
That is fixed.
The script was improved by adding a variable, cleaning up and hence
also fixing shellcheck warnings.

eldy committed:
Update jstz

eldy committed:
QUAL: Remove the external js library jstimezonedetect

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Clean dolibarr about page

eldy committed:
Merge pull request #27509 from frederic34/patch-1

fix phpstan

eldy committed:
Merge pull request #27511 from frederic34/patch-5

fix phpstan

eldy committed:
Merge pull request #27513 from mdeweerd/fix/github_lines_byuser

Fix: github_lines_byuser.sh - resolve issue with iconv

eldy committed:
Merge pull request #27510 from frederic34/patch-3

fix phpstan

eldy committed:
Merge pull request #27512 from frederic34/patch-7

fix phpstan

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

Commit:
File not joined : slash is missing in case DOL_DATA_ROOT doesn't end with slash

eldy committed:
Swith to level 3 of phpstan

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

eldy committed:
Fix phpstan

eldy committed:
Fix phpstan

dependabot[bot] committed:
Bump actions/upload-artifact from 3 to 4

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 3 to 4.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/setup-python from 4 to 5

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 4 to 5.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/setup-node from 3 to 4

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 3 to 4.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

W1W1-M committed:
Fixed selected default value

frederic34 committed:
fix phpstan

W1W1-M committed:
Merge branch 'develop' into fix-resource-phpstan

W1W1-M committed:
Updated PHPDoc

W1W1-M committed:
Merge remote-tracking branch 'origin/fix-resource-phpstan' into fix-resource-phpstan

# Conflicts:
#	htdocs/resource/class/html.formresource.class.php

eldy committed:
Merge pull request #27526 from frederic34/patch-12

fix phpstan

eldy committed:
Merge pull request #27520 from frederic34/patch-10

fix phpstan

eldy committed:
Clean code

eldy committed:
Merge pull request #27527 from W1W1-M/fix-resource-phpstan

Fix resource phpstan

eldy committed:
Merge pull request #27514 from frederic34/patch-8

fix phpstan

eldy committed:
Merge pull request #27515 from frederic34/patch-1

fix phpstan

eldy committed:
Merge pull request #27516 from frederic34/patch-3

fix phpstan

eldy committed:
Merge pull request #27521 from libremaster/develop

File not joined : slash is missing in case DOL_DATA_ROOT doesn't end with a slash

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #27517 from frederic34/patch-5

fix phpstan

eldy committed:
Merge pull request #27518 from frederic34/patch-7

fix phpstan

eldy committed:
Merge pull request #27523 from Dolibarr/dependabot/github_actions/actions/upload-artifact-4

Bump actions/upload-artifact from 3 to 4

eldy committed:
Merge pull request #27524 from Dolibarr/dependabot/github_actions/actions/setup-python-5

Bump actions/setup-python from 4 to 5

eldy committed:
Merge pull request #27525 from Dolibarr/dependabot/github_actions/actions/setup-node-4

Bump actions/setup-node from 3 to 4

vmaury committed:
Fix errors in supplier order odt template

eldy committed:
NEW Upgrade Sabre lib to 4.6.0

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix phpstan

eldy committed:
Merge pull request #27531 from vmaury/Fix_error_in_supplier_order_odt_template

Fix errors in supplier order odt template

Commit:
NEW: cronjob: deny to launch a method from a deactivated module

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
FIX: typo: Swith => Switch

Commit:
FIX: Remove blank line

eldy committed:
Merge branch '19.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy commented on htdocs/core/class/html.form.class.php:
```diff
@@ -6460,7 +6460,7 @@ public function load_cache_vatrates($country_code)
 	 *                         Si le (pays vendeur = pays acheteur) alors la TVA par defaut=TVA du produit vendu. Fin de regle.
 	 *                         Si (vendeur et acheteur dans Communaute europeenne) et bien vendu = moyen de transports neuf (auto, bateau, avion), TVA par defaut=0 (La TVA doit etre paye par l'acheteur au centre d'impots de son pays et non au vendeur). Fin de regle.
-	 *                         Si vendeur et acheteur dans Communauté européenne et acheteur= particulier alors TVA par défaut=TVA du produit vendu. Fin de règle.
+	 *                         Si vendeur et acheteur dans Communauté européenne et acheteur= particulier alors TVA par défaut=TVA du pays acheteur. Fin de règle.
```
I think the term "du produit vendu" was the good one so if we sell wine in a restaurant it may be vat 10% and if meet, it may be 20%).  We use the vat of selling country only if product is unknow.

comaiteseb replied:
I understand what you mean but, there is a problèm with the main reason why this rule was introduced.
If you sell meat to an EEC country, you have to apply the VAT of this country. 
This calls into question the structure of VAT rates because there is a lack of a notion of “VAT class” in order to be able to transpose VAT between different countries. ex: VAT 20% (FR) <=> VAT 19%(DE)

eldy commented on htdocs/core/class/html.form.class.php:
```diff
@@ -6511,19 +6511,19 @@ public function load_tva($htmlname = 'tauxtva', $selectedrate = '', $societe_ven
 				if (is_numeric($type)) {
 					if ($type == 1) { // We know product is a service
-						$code_country .= ",'" . $societe_acheteuse->country_code . "'";
+						$code_country = "'" . $societe_acheteuse->country_code . "'";
```
When SERVICE_ARE_ECOMMERCE_200238EC is set, we want to have ALSO in the combo list the vat rate of the purchaser country because there is a lot of exception and restricting the list to purchaser country only will block user if he need to fallback to its own vat rate. This is why we did a concat .=

comaiteseb replied:
Ok, I think it's the right way. But where should I document this new value for SERVICE_ARE_ECOMMERCE_200238EC ?

eldy commented on htdocs/core/class/html.form.class.php:
```diff
@@ -6511,19 +6511,19 @@ public function load_tva($htmlname = 'tauxtva', $selectedrate = '', $societe_ven
 		if (getDolGlobalString('SERVICE_ARE_ECOMMERCE_200238EC')) {    // If option to have vat for end customer for services is on
 			require_once DOL_DOCUMENT_ROOT . '/core/lib/company.lib.php';
-			if (!isInEEC($societe_vendeuse) && (!is_object($societe_acheteuse) || (isInEEC($societe_acheteuse) && !$societe_acheteuse->isACompany()))) {
+			if (isInEEC($societe_vendeuse) && isInEEC($societe_acheteuse) && !$societe_acheteuse->isACompany()) {
```
This seems the fix and looks OK.

eldy committed:
Fix phpcs

Commit:
Simpler version to check if a module is enabled

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

thibdrev committed:
phpstan

Function dol_get_first_day() should return int but returns string.
Function getWeekNumber() should return int but returns string.

lamrani002 committed:
add function for list event for user

frederic34 committed:
Update .pre-commit-config.yaml

eldy committed:
Doc

eldy committed:
Update usergroups.lib.php

eldy committed:
Merge branch 'develop' into updateEventAgendaOnUser

eldy committed:
Fix case

eldy committed:
Merge branch 'develop' into patch-3

frederic34 committed:
Update .pre-commit-config.yaml

eldy committed:
Merge pull request #27546 from frederic34/patch-3

Update .pre-commit-config.yaml

eldy committed:
Merge branch 'develop' into updateEventAgendaOnUser

eldy committed:
Merge pull request #27537 from libremaster/develop

NEW: cronjob: deny to launch a method from a deactivated module

eldy committed:
Merge pull request #27099 from altairisfr/ordermode

add constant SUPPLIER_ORDER_MODE_OPTIONAL

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

frederic34 committed:
fix phpstan

thibdrev committed:
phpstan

Function dol_get_first_day() should return int but returns string.
Function dol_get_last_day() should return int but returns string.

frederic34 committed:
Update .pre-commit-config.yaml

eldy committed:
Merge pull request #27547 from frederic34/patch-3

fix phpstan

eldy committed:
Merge pull request #27550 from frederic34/patch-8

fix phpstan

eldy committed:
Merge pull request #27549 from frederic34/patch-7

fix phpstan

eldy committed:
Merge pull request #27548 from frederic34/patch-5

fix phpstan

eldy committed:
Fix doxygen

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #27545 from lamrani002/updateEventAgendaOnUser

NEW functionality for list event for user

eldy committed:
Merge pull request #27551 from frederic34/patch-10

fix phpstan

eldy committed:
Merge pull request #27552 from frederic34/patch-12

Update .pre-commit-config.yaml

eldy committed:
Merge pull request #27544 from thibdrev/patch-1

fix phpstan

eldy committed:
Update user.php

eldy committed:
Update user.php

eldy committed:
Merge pull request #25740 from atm-thibaultf/NEW_hide_external_user_in_combobox

NEW : hide external user in combobox
